src/mpyl/steps/deploy/k8s/chart.py

```diff
@@ -481,6 +485,23 @@ def _get_job(self) -> Job:
             raise AttributeError("deployment.kubernetes.job field should be set")
         return job
 
+    def _create_secret_env_vars(
+        self, secret_list: list[KeyValueProperty]
+    ) -> list[V1EnvVar]:
+        return list(
+            map(
+                lambda e: V1EnvVar(
+                    name=e.key,
+                    value_from=V1EnvVarSource(
+                        secret_key_ref=V1SecretKeySelector(
+                            key=e.key, name=self.release_name, optional=False
+                        )
+                    ),
+                ),
+                secret_list,
+            )
+        )
```

Comment:
I like this style better than`list(map(...))` but leave it up to you :)
```suggestion
        return [
            V1EnvVar(
                name=e.key,
                value_from=V1EnvVarSource(
                    secret_key_ref=V1SecretKeySelector(
                        key=e.key, name=self.release_name, optional=False
                    )
                ),
            )
            for e in secret_list
        ]
```